pkgs/top-level/coq-packages.nix

```diff
@@ -64,23 +70,22 @@ let
       VST = callPackage ../development/coq-modules/VST {
         compcert = compcert.override { version = "3.7"; };
       };
-
-      filterPackages = filterCoqPackages;
+      filterPackages = doesFilter: if doesFilter then filterCoqPackages self else self;
     };
 
-  filterCoqPackages = coq: set:
+  filterCoqPackages = set:
     lib.listToAttrs (
-      lib.concatMap (name:
-        let v = set.${name}; in
-        let p = v.compatibleCoqVersions or (_: true); in
-        lib.optional (p coq.coq-version)
-          (lib.nameValuePair name (
-            if lib.isAttrs v && v.recurseForDerivations or false
-            then filterCoqPackages coq v
-            else v))
+      lib.concatMap (name: let v = set.${name} or null; in
+          lib.optional (! v.meta.coqFilter or false)
```

Comment:
Is the `meta.coqFilter` attribute undocumented on purpose? I.e., not to be used by the casual contributor (contrary to `compatibleCoqVersions`).

Reply:
Indeed this is an internal variable, maybe I should even raise an error if one tries to override it.

Reply:
The way to declare a package incompatible and withdraw it from `coqPackages` is tp set `defaultVersion` to `null`.